Failing builds with CALT

It happens occasionally (so far observed only for planet OSM file) that the landmark preprocessing step for CALT fails with the following:

```13 Jun 10:56:20 ERROR [routing.RoutingProfileManager] - java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at heigit.ors.routing.RoutingProfileManager.initialize(RoutingProfileManager.java:218)
	at heigit.ors.routing.RoutingProfileManager.getInstance(RoutingProfileManager.java:66)
	at heigit.ors.servlet.listeners.ORSInitContextListener.lambda$contextInitialized$0(ORSInitContextListener.java:43)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at heigit.ors.routing.graphhopper.extensions.core.CoreLMAlgoFactoryDecorator.loadOrDoWork(CoreLMAlgoFactoryDecorator.java:278)
	at heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.loadOrPrepareCoreLM(ORSGraphHopper.java:625)
	at heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.postProcessing(ORSGraphHopper.java:541)
	at com.graphhopper.GraphHopper.process(GraphHopper.java:719)
	at com.graphhopper.GraphHopper.importOrLoad(GraphHopper.java:688)
	at heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.importOrLoad(ORSGraphHopper.java:158)
	at heigit.ors.routing.RoutingProfile.initGraphHopper(RoutingProfile.java:185)
	at heigit.ors.routing.RoutingProfile.<init>(RoutingProfile.java:130)
	at heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:37)
	at heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:21)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLMAlgoFactoryDecorator.loadOrDoWork(CoreLMAlgoFactoryDecorator.java:274)
	... 15 more
Caused by: java.lang.NullPointerException
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage.expandEdge(CoreLandmarkStorage.java:155)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage.expandEdge(CoreLandmarkStorage.java:178)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage.expandEdge(CoreLandmarkStorage.java:171)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage.expandEdge(CoreLandmarkStorage.java:171)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage.access$100(CoreLandmarkStorage.java:56)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage$1.calcWeight(CoreLandmarkStorage.java:121)
	at com.graphhopper.routing.DijkstraBidirectionRef.fillEdges(DijkstraBidirectionRef.java:172)
	at com.graphhopper.routing.DijkstraBidirectionRef.fillEdgesFrom(DijkstraBidirectionRef.java:132)
	at com.graphhopper.routing.AbstractBidirAlgo.runAlgo(AbstractBidirAlgo.java:67)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage$CoreLandmarkExplorer.runAlgo(CoreLandmarkStorage.java:848)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage.createLandmarksForSubnetwork(CoreLandmarkStorage.java:361)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLandmarkStorage.createLandmarks(CoreLandmarkStorage.java:270)
	at heigit.ors.routing.graphhopper.extensions.core.PrepareCoreLandmarks.doWork(PrepareCoreLandmarks.java:134)
	at heigit.ors.routing.graphhopper.extensions.core.CoreLMAlgoFactoryDecorator$1.run(CoreLMAlgoFactoryDecorator.java:264)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	... 6 more```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed the wrong gpx header for api v2 (#496)
 - Make sure external storages contain entries for all edge IDs (#535)
 - Check if BordersStorage exists before calling it in AvoidBordersCoreEdgeFilter
+- Take into account shortcut direction in LM selection weighting (#550)
 ### Changed
 - Moved walking and hiking flag encoders to the ORS core system (#440)
 - Remove route optimization code (#499)

--- a/openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/core/CoreLandmarkStorage.java
+++ b/openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/core/CoreLandmarkStorage.java
@@ -118,7 +118,7 @@ public class CoreLandmarkStorage implements Storable<LandmarkStorage>{
                     if (res >= Double.MAX_VALUE)
                         return Double.POSITIVE_INFINITY;
 
-                    expandEdge(tmp, false);
+                    expandEdge(tmp, reverse);
 
                     return count;
                 }
@@ -161,22 +161,25 @@ public class CoreLandmarkStorage implements Storable<LandmarkStorage>{
         int skippedEdge2 = mainEdgeState.getSkippedEdge2();
         int from = mainEdgeState.getBaseNode(), to = mainEdgeState.getAdjNode();
 
+        if (reverse) {
+            int tmp = from;
+            from = to;
+            to = tmp;
+        }
 
+        CHEdgeIteratorState iter = core.getEdgeIteratorState(skippedEdge1, from);
+        boolean empty = iter == null;
+        if (empty)
+            iter =  core.getEdgeIteratorState(skippedEdge2, from);
 
-            CHEdgeIteratorState iter = core.getEdgeIteratorState(skippedEdge1, from);
-            boolean empty = iter == null;
-            if (empty)
-                iter =  core.getEdgeIteratorState(skippedEdge2, from);
+        expandEdge(iter, true);
 
-            expandEdge(iter, true);
+        if (empty)
+            iter =  core.getEdgeIteratorState(skippedEdge1, to);
+        else
+            iter =  core.getEdgeIteratorState(skippedEdge2, to);
 
-            if (empty)
-                iter =  core.getEdgeIteratorState(skippedEdge1, to);
-            else
-                iter =  core.getEdgeIteratorState(skippedEdge2, to);
-
-            expandEdge(iter, false);
-
+        expandEdge(iter, false);
     }
 
     /**

